### PR TITLE
Add document-link elements within iati-activity/result

### LINKED ISSUE #359: Add new document-link element (as a child of various result elements)
Add a new `document-link` element at the following locations:
- `iati-activities/iati-activity/result/document-link`
- `iati-activities/iati-activity/result/indicator/document-link`
- `iati-activities/iati-activity/result/indicator/baseline/document-link`
- `iati-activities/iati-activity/result/indicator/period/target/document-link`
- `iati-activities/iati-activity/result/indicator/period/actual/document-link`

The element will have an occurrence rule of `0..*` at all of the above locations, and will have following definition:

> A link to an online, publicly accessible web page or document expanding on the result.

This change replicates the existing `iati-activities/iati-activity/document-link` element, with the addition of the new description element to be added in #357. 

**Tasks:**

- [x] Check for other instances of the `document-link` element (across the [activity](https://github.com/IATI/IATI-Schemas/blob/version-2.02/iati-activities-schema.xsd#L2749-L2834)) and [organisation](https://github.com/IATI/IATI-Schemas/blob/version-2.02/iati-organisations-schema.xsd#L758-L865) schemas. Create a new common element. - Done in IATI/IATI-Schemas#389
- [x] Add tests for all elements above - Done in IATI/IATI-Schemas#389
- [x] Add `iati-activities/iati-activity/result/document-link` - Done in IATI/IATI-Schemas#389
- [x] Add `iati-activities/iati-activity/result/indicator/document-link` - Done in IATI/IATI-Schemas#389
- [x] Add `iati-activities/iati-activity/result/indicator/baseline/document-link` - Done in IATI/IATI-Schemas#389
- [x] Add `iati-activities/iati-activity/result/indicator/period/target/document-link` - Done in IATI/IATI-Schemas#389
- [x] Add `iati-activities/iati-activity/result/indicator/period/actual/document-link` - Done in IATI/IATI-Schemas#389

- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/document-link/@format` (to reference this attribute to the `FileFormat` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/document-link/title/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/document-link/description/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/document-link/category/@code` (to reference this attribute to the `DocumentCategory` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/document-link/language/@code` (to reference this attribute to the `Language` codelist) - Done in IATI/IATI-Codelists#123

- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/document-link/@format` (to reference this attribute to the `FileFormat` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/document-link/title/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/document-link/description/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/document-link/category/@code` (to reference this attribute to the `DocumentCategory` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/document-link/language/@code` (to reference this attribute to the `Language` codelist) - Done in IATI/IATI-Codelists#123

- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/baseline/document-link/@format` (to reference this attribute to the `FileFormat` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/baseline/document-link/title/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/baseline/document-link/description/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/baseline/document-link/category/@code` (to reference this attribute to the `DocumentCategory` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/baseline/document-link/language/@code` (to reference this attribute to the `Language` codelist) - Done in IATI/IATI-Codelists#123

- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/target/document-link/@format` (to reference this attribute to the `FileFormat` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/target/document-link/title/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/target/document-link/description/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/target/document-link/category/@code` (to reference this attribute to the `DocumentCategory` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/target/document-link/language/@code` (to reference this attribute to the `Language` codelist) - Done in IATI/IATI-Codelists#123

- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/actual/document-link/@format` (to reference this attribute to the `FileFormat` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/actual/document-link/title/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/actual/document-link/description/narrative/@xml:lang` (to reference this attribute to the `Language` codelist) - Not needed, as [paths for all `xml:lang` attributes already exist](https://github.com/IATI/IATI-Codelists/blob/694518226f67a5de88df251e27477d49cf29b4de/mapping.xml#L296-L299)
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/actual/document-link/category/@code` (to reference this attribute to the `DocumentCategory` codelist) - Done in IATI/IATI-Codelists#123
- [x] Update Codelist mapping file for `iati-activities/iati-activity/result/indicator/period/actual/document-link/language/@code` (to reference this attribute to the `Language` codelist) - Done in IATI/IATI-Codelists#123

- [x] Merge https://github.com/IATI/IATI-Codelists/pull/123 to the correct branch - Done in IATI/IATI-Codelists#132

### PR DESCRIPTION
**Pull request by dalepotter**
Fixes #359

### CODE REVIEW

File: tests/should-fail/iati-activities/iati-activity/result/document-link/description/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <description additional-attribute="xx"><!-- FAIL: An addiitonal attribute added. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/description/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml]: typo: addiitonal -> `additional`

File: tests/should-fail/iati-activities/iati-activity/result/document-link/category/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx" additional-attribute="xx" /><!-- FAIL: An addiitonal attribute added. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/category/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml]: typo on additional here also

File: tests/should-fail/iati-activities/iati-activity/result/document-link/description/err-not-iati-xml-missing-required-element_document-link-with-multiple-description-elements.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <description>
+          <narrative>Sample text</narrative>
+        </description>
+        <description><!-- FAIL: Optional element more than the allowed number of times. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/description/err-not-iati-xml-missing-required-element_document-link-with-multiple-description-elements.xml]: rephrase: `Optional element present more than the permitted number of times.`

File: tests/should-fail/iati-activities/iati-activity/result/document-link/document-date/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <document-date iso-date="xx" additional-attribute="xx"/><!-- FAIL: An additional attribute added. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/document-date/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml]: Rephrase: `Non-permitted additional attribute present.`

File: tests/should-fail/iati-activities/iati-activity/result/document-link/document-date/err-not-iati-xml-incorrect-datatype.xml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <document-date iso-date="xx"/><!-- FAIL: Not an XML date type. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/document-date/err-not-iati-xml-incorrect-datatype.xml]: rephrase: `Not a valid xsd:date.`

File: tests/should-fail/iati-activities/iati-activity/result/document-link/document-date/err-not-iati-xml-missing-required-element_multiple-elements-present.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <document-date iso-date="2000-01-01"/>
+        <document-date iso-date="2000-01-02"/><!-- FAIL: An addiitonal element is added. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/document-date/err-not-iati-xml-missing-required-element_multiple-elements-present.xml]: The typo returns! addiitonal -> `additional`

File: tests/should-fail/iati-activities/iati-activity/result/document-link/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    <activity-status code="xx"/>
+    <activity-date type="xx" iso-date="2013-11-27"/>
+    <activity-date type="xx" iso-date="2013-11-27">
+      <narrative>Some stuff ere</narrative>
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx" additional-attribute="xx"><!-- FAIL: Optional document-link element added with required child attributes/elements but an addiitonal attribute added. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml]: and here too `additional`

File: tests/should-fail/iati-activities/iati-activity/result/document-link/language/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <language additional-attribute="xx"/><!-- FAIL: An addiitonal attribute added. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/language/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml]: And here

File: tests/should-fail/iati-activities/iati-activity/result/document-link/title/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    <activity-date type="xx" iso-date="2013-11-27"/>
+    <activity-date type="xx" iso-date="2013-11-27">
+      <narrative>Some stuff ere</narrative>
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title additional-attribute="xx"><!-- FAIL: An addiitonal attribute added. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/title/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml]: and here

File: tests/should-fail/iati-activities/iati-activity/result/document-link/title/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx" />
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/title/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml]: remove unnecessary whitespace

File: tests/should-fail/iati-activities/iati-activity/result/document-link/title/narrative/err-not-iati-xml-missing-required-element_no-narrative-element.xml [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    <activity-date type="xx" iso-date="2013-11-27">
+      <narrative>Some stuff ere</narrative>
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <!-- FAIL: Required `title` element added, but without the required 'narrative' child element. -->
```
Comment (allthatilk) [inline tests/should-fail/iati-activities/iati-activity/result/document-link/title/narrative/err-not-iati-xml-missing-required-element_no-narrative-element.xml]: Remove whitespace

File: tests/should-pass/iati-activities/iati-activity/result/document-link/document-date/element-present-with-namespace-attribute.xml [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <document-date iso-date="2000-01-01" example-namespace:namespace-attribute="xx"/><!-- PASS: Optional element added with required child element, and namespace attribute. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/document-link/document-date/element-present-with-namespace-attribute.xml]: remove comma before `and`

File: tests/should-pass/iati-activities/iati-activity/result/document-link/element-present-with-namespace-attribute.xml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    <activity-status code="xx"/>
+    <activity-date type="xx" iso-date="2013-11-27"/>
+    <activity-date type="xx" iso-date="2013-11-27">
+      <narrative>Some stuff ere</narrative>
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx" example-namespace:namespace-attribute="xx"><!-- PASS: Optional element added with required child elements, and namespace attribute. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/document-link/element-present-with-namespace-attribute.xml]: remove comma before `and` here and elsewhere

File: tests/should-pass/iati-activities/iati-activity/result/document-link/language/element-present-with-namespace-attribute.xml [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx"/>
+        <language code="xx" example-namespace:namespace-attribute="xx"/><!-- PASS: Optional element added with required child element, and namespace attribute. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/document-link/language/element-present-with-namespace-attribute.xml]: here

File: tests/should-pass/iati-activities/iati-activity/result/document-link/category/element-present-with-namespace-attribute.xml [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <category code="xx" example-namespace:namespace-attribute="xx"/><!-- PASS: Element added with required child element, and namespace attribute. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/document-link/category/element-present-with-namespace-attribute.xml]: here

File: tests/should-pass/iati-activities/iati-activity/result/document-link/description/element-present-with-namespace-attribute.xml [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <description example-namespace:namespace-attribute="xx"><!-- PASS: Optional element added with required child element, and namespace attribute. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/document-link/description/element-present-with-namespace-attribute.xml]: here

File: tests/should-pass/iati-activities/iati-activity/result/document-link/description/narrative/element-present-with-namespace-attribute.xml [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <description>
+          <narrative example-namespace:namespace-attribute="xx">Sample text</narrative><!-- PASS: Required element added, and namespace attribute. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/document-link/description/narrative/element-present-with-namespace-attribute.xml]: here

File: tests/should-pass/iati-activities/iati-activity/result/document-link/title/element-present-with-namespace-attribute.xml [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+    <activity-date type="xx" iso-date="2013-11-27"/>
+    <activity-date type="xx" iso-date="2013-11-27">
+      <narrative>Some stuff ere</narrative>
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title example-namespace:namespace-attribute="xx"><!-- PASS: Optional element added with required child element, and namespace attribute. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/document-link/title/element-present-with-namespace-attribute.xml]: here

File: tests/should-pass/iati-activities/iati-activity/result/document-link/title/narrative/element-present-with-namespace-attribute.xml [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+    <activity-date type="xx" iso-date="2013-11-27">
+      <narrative>Some stuff ere</narrative>
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative example-namespace:namespace-attribute="xx">Sample text</narrative><!-- PASS: Required element added, and namespace attribute. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/document-link/title/narrative/element-present-with-namespace-attribute.xml]: here

File: tests/should-pass/iati-activities/iati-activity/result/document-link/title/narrative/multiple-elements-present.xml [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+      <narrative>Some stuff ere</narrative>
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <document-link url="xx" format="xx">
+        <title>
+          <narrative>Sample text</narrative>
+          <narrative>Sample text</narrative><!-- PASS: Multiple elements present. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/document-link/title/narrative/multiple-elements-present.xml]: rephrase: `element present multiple times in context`

**Review by allthatilk — CHANGES_REQUESTED** [review]
All just small typo things

File: tests/should-fail/iati-activities/iati-activity/result/document-link/title/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml (hunk 10 above)
Comment (dalepotter) [reply tests/should-fail/iati-activities/iati-activity/result/document-link/title/err-not-iati-xml-forbidden-attribute_non-permitted-attribute-present.xml]: Is there unnecessary whitespace - I can't see any...

image

File: tests/should-fail/iati-activities/iati-activity/result/document-link/title/narrative/err-not-iati-xml-missing-required-element_no-narrative-element.xml (hunk 11 above)
Comment (dalepotter) [reply tests/should-fail/iati-activities/iati-activity/result/document-link/title/narrative/err-not-iati-xml-missing-required-element_no-narrative-element.xml]: Is the suggestion that this should be:

```
<title/><!-- FAIL: Required `title` element added, but without the required 'narrative' child element. -->
```

??

File: tests/should-pass/iati-activities/iati-activity/result/document-link/category/element-present-with-namespace-attribute.xml (hunk 15 above)
Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/document-link/category/element-present-with-namespace-attribute.xml]: what's here? Guessing the unnecessary comma?

File: tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/language/element-present.xml [hunk 21]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0"?>
+
+<iati-activities version="xx">
+  <iati-activity xmlns:example-namespace="http://example.org/example-namespace/ns#"><!-- Include namespace declaration. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/language/element-present.xml]: should this be here?

File: tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/document-date/element-present.xml [hunk 22]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0"?>
+
+<iati-activities version="xx">
+  <iati-activity xmlns:example-namespace="http://example.org/example-namespace/ns#"><!-- Include namespace declaration. -->
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/document-date/element-present.xml]: here?

File: tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-attribute.xml [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+      </title>
+      <indicator measure="xx" ascending="0">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <document-link url="xx" format="xx">
+          <title example-namespace:namespace-attribute="xx"><!-- PASS: Optional element added with required child element and namespace attribute. -->
+            <narrative>Sample text</narrative>
+          </title>
+          <category code="xx" />
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-attribute.xml]: whitespace, might want to do a check throughout as it always seems to be with the code attribute.

File: tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-subelement.xml [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+      <indicator measure="xx" ascending="0">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <document-link url="xx" format="xx">
+          <title>
+            <narrative>Sample text</narrative>
+            <example-namespace:namespace-element></example-namespace:namespace-element> <!-- PASS: Optional namespace element included. -->
+          </title>
+          <category code="xx" />
```
Comment (allthatilk) [inline tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-subelement.xml]: whitespace

**Review by allthatilk — CHANGES_REQUESTED** [review]

File: tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/document-date/element-present.xml (hunk 22 above)
Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/document-date/element-present.xml]: Fixed in 4a91fd0

File: tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/language/element-present.xml (hunk 21 above)
Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/language/element-present.xml]: Fixed in 4a91fd0

File: tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-attribute.xml (hunk 23 above)
Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-attribute.xml]: This was resolved in caf1ce7.

File: tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-subelement.xml (hunk 24 above)
Comment (dalepotter) [reply tests/should-pass/iati-activities/iati-activity/result/indicator/document-link/title/element-present-with-namespace-subelement.xml]: This was resolved in caf1ce7.

**Comment by dalepotter** [comment]
@allthatilk this is now ready for re-review (checking the three most recent commits to this branch should suffice).

**Review by allthatilk — APPROVED** [review]
